### PR TITLE
Adds support for v1.9x

### PR DESCRIPTION
**Pull request by 6za**
Details at: https://github.com/kubefirst/kubefirst/issues/253

This PR update metaphor to support github actions, and be feature compatible to the gitlab version. 

Based on guide: https://docs.github.com/en/actions/migrating-to-github-actions/migrating-from-gitlab-cicd-to-github-actions

### CODE REVIEW

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+            -p environment="staging" \
+            -p gitRepoUrl="${GITREPO_URL}" \
+            -p shortSha="${GITHUB_SHA:7}" \
+            --wait --log   
+      - run: echo "🍏 This job's status is ${{ job.status }}."              
+  release:
+    needs: staging  
+    runs-on: self-hosted
+    steps:
+      - name: Setup Runner for Argo
```
Comment (6za) [inline .github/workflows/main.yaml:130]: This Step will in the future become part of runner:https://github.com/kubefirst/kubefirst/issues/397

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+            -p shortSha="${GITHUB_SHA:7}" \
+            --wait --log  
+      - run: echo "🍏 This job's status is ${{ job.status }}."              
+  staging:
+    needs: development
+    runs-on: self-hosted
+    steps:
+      - name: Setup Runner for Argo
+        run: |
+            cd $HOME
```
Comment (6za) [inline .github/workflows/main.yaml:97]: Refers to: https://github.com/kubefirst/kubefirst/issues/397

File: .github/workflows/main.yaml [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+            --wait --log
+      - run: echo "🍏 This job's status is ${{ job.status }}."              
+  development:
+    needs: publish
+    runs-on: self-hosted
+    steps:
+      - name: Setup Runner for Argo
+        run: |
+            cd $HOME
+            echo "Install argo"
```
Comment (6za) [inline .github/workflows/main.yaml:63]: refers to: https://github.com/kubefirst/kubefirst/issues/397

File: .github/workflows/build.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+jobs:
+  build:
+    runs-on: self-hosted
+    steps:
+      - name: Setup Runner for Argo
+        run: |
+            cd $HOME
+            echo "Install argo"
+            # Download the binary
+            curl -sLO https://github.com/argoproj/argo-workflows/releases/download/${ARGO_VERSION}/argo-linux-amd64.gz
```
Comment (6za) [inline .github/workflows/build.yaml:23]: refers to: https://github.com/kubefirst/kubefirst/issues/397

File: .github/workflows/main.yaml [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+jobs:
+  publish:
+    runs-on: self-hosted
+    steps:
+      - name: Setup Runner for Argo
+        run: |
+            cd $HOME
+            echo "Install argo"
+            # Download the binary
+            curl -sLO https://github.com/argoproj/argo-workflows/releases/download/${ARGO_VERSION}/argo-linux-amd64.gz
```
Comment (6za) [inline .github/workflows/main.yaml:27]: refers to: https://github.com/kubefirst/kubefirst/issues/397

File: .github/workflows/main.yaml [hunk 6]
```diff
@@ -0,0 +1,160 @@
+name:  Release Main
+env:
+  ARGO_NAMESPACE: argo
+  ARGO_VERSION: v3.1.11
+# this needs to be a template
+  GITREPO_URL: ssh://git@<GITHUB_HOST>/<GITHUB_OWNER>/metaphor.git
+  CI_PROJECT_NAME: metaphor
```
Comment (6za) [inline .github/workflows/main.yaml:7]: Github variables don't work at this level, this were kept like this to make easier to read the code and simplify use on the other metaphors repo.

**Review by converge — APPROVED** [review]

**Pull request merged**